src/service/handlers/prepared.rs

```diff
@@ -276,21 +282,40 @@ pub(crate) async fn do_put_prepared_statement_update(
         ));
     }
 
-    let parameter_sets = SwanFlightSqlService::collect_parameter_sets(request).await?;
+    // Collect RecordBatches first - we can only consume the stream once
+    let batches = SwanFlightSqlService::collect_record_batches(request).await?;
+    
+    // No batches provided - execute with empty parameters
+    if batches.is_empty() {
+        info!(
+            handle = %handle,
+            sql = %sql,
+            "executing prepared statement update without parameters"
+        );
 
-    let parameter_set_count = parameter_sets.len();
+        let session_clone = Arc::clone(&session);
+        let affected_rows = tokio::task::spawn_blocking(move || {
+            SwanFlightSqlService::execute_statement_batches(&sql, &[], &session_clone)
+        })
+        .await
+        .map_err(SwanFlightSqlService::status_from_join)?
+        .map_err(SwanFlightSqlService::status_from_error)?;
 
-    info!(
-        handle = %handle,
-        sql = %sql,
-        parameter_sets = parameter_set_count,
-        "executing prepared statement update"
-    );
+        info!(
+            handle = %handle,
+            affected_rows,
+            "prepared statement update complete"
+        );
+        return Ok(affected_rows);
+    }
 
+    // Use appender optimization via session layer
     let session_clone = Arc::clone(&session);
-
+    
     let affected_rows = tokio::task::spawn_blocking(move || {
-        SwanFlightSqlService::execute_statement_batches(&sql, &parameter_sets, &session_clone)
+        session_clone
+            .execute_with_record_batches(&sql, batches)
```

Comment:
if above code use sql_parser, can we just pass table_name and batches into this method, so that we don't need to parse the sql again. I believe execute_with_record_batches method will be greatly simplified!